[jit] Make `load()` create only one CU

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22727 [jit] Make `load()` create only one CU**

Now, `torch.jit.load()` and variants will create a single compilation
unit, and compile everything with the appropriate qualified names into
it.

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

Differential Revision: [D16197603](https://our.internmc.facebook.com/intern/diff/D16197603)

suo committed:
[wip] load to a single CU

suo committed:
Update on "[wip] load to a single CU"

[wip] load to a single CU

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

suo committed:
Update on "[wip] load to a single CU"

[wip] load to a single CU

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

suo committed:
Update on "[wip] load to a single CU"

[jit] Make `load()` create only one CU

Now, `torch.jit.load()` and variants will create a single compilation
unit, and compile everything with the appropriate qualified names into
it.

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

zdevito commented on torch/csrc/jit/script/module.cpp:
```diff
@@ -12,6 +12,35 @@ namespace torch {
+Module::Module(
+    c10::QualifiedName class_name,
+    std::shared_ptr<CompilationUnit> cu) {
+  auto cls = ClassType::create(
```
Shouldn't this be a call to create_module_object as well? Only difference is the cu

suo committed:
Update on "[jit] Make `load()` create only one CU"

[jit] Make `load()` create only one CU

Now, `torch.jit.load()` and variants will create a single compilation
unit, and compile everything with the appropriate qualified names into
it.

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

suo committed:
Update on "[jit] Make `load()` create only one CU"

[jit] Make `load()` create only one CU

Now, `torch.jit.load()` and variants will create a single compilation
unit, and compile everything with the appropriate qualified names into
it.

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

suo committed:
Update on "[jit] Make `load()` create only one CU"

[jit] Make `load()` create only one CU

Now, `torch.jit.load()` and variants will create a single compilation
unit, and compile everything with the appropriate qualified names into
it.

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

suo committed:
Update on "[jit] Make `load()` create only one CU"

[jit] Make `load()` create only one CU

Now, `torch.jit.load()` and variants will create a single compilation
unit, and compile everything with the appropriate qualified names into
it.

gh-metadata: pytorch pytorch 22727 gh/suo/92/head

suo committed:
Update on "[jit] Make `load()` create only one CU"

[jit] Make `load()` create only one CU

Now, `torch.jit.load()` and variants will create a single compilation
unit, and compile everything with the appropriate qualified names into
it.

gh-metadata: pytorch pytorch 22727 gh/suo/92/head
